File: Blind-75/TwoPointers/ContainerWithTheMostWater/pb-11.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0; int r = height.length-1;
        int res = 0;
        while (l <= r){
            int water = Math.min(height[l], height[r]) * (r-l);
            res = Math.max(res, water);
            if (height[l] < height[r]){
                l++;
            } else {
                r--;
            }
        }
        return res;
    }
}

File: Blind-75/TwoPointers/ContainerWithTheMostWater/pb-11.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        # area = min(h1, h2) * idx(h2)-idx(h1)
        l, r = 0, len(height)-1
        res = 0
        while l <= r:
            area = (r-l) * min(height[l], height[r])
            res = max(res, area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return res   